RubberduckTests/Inspections/DefTypeStatementInspectionTests.cs

```diff
@@ -0,0 +1,326 @@
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void DefType_Date()
+        {
+            const string inputCode =
+@"DefDate H
+Public Function hFoo()
+End Function";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DefTypeStatementInspection(state);
+                var inspector = InspectionsHelper.GetInspector(inspection);
+                var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void DefType_String()
+        {
+            const string inputCode =
+@"DefStr I
+Public Function iFoo()
+End Function";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DefTypeStatementInspection(state);
+                var inspector = InspectionsHelper.GetInspector(inspection);
+                var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void DefType_Object()
+        {
+            const string inputCode =
+@"DefObj J
+Public Function jFoo()
+End Function";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DefTypeStatementInspection(state);
+                var inspector = InspectionsHelper.GetInspector(inspection);
+                var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
+
+        [Test]
+        [Category("Inspections")]
+        public void DefType_Variant()
+        {
+            const string inputCode =
+@"DefVar K
+Public Function kFoo()
+End Function";
+
+            var vbe = MockVbeBuilder.BuildFromSingleStandardModule(inputCode, out _);
+            using (var state = MockParser.CreateAndParse(vbe.Object))
+            {
+                var inspection = new DefTypeStatementInspection(state);
+                var inspector = InspectionsHelper.GetInspector(inspection);
+                var inspectionResults = inspector.FindIssuesAsync(state, CancellationToken.None).Result;
+
+                Assert.AreEqual(1, inspectionResults.Count());
+            }
+        }
```

Comment:
Make these tests be data-driven. You can do something like: `[TestCase("DefVar")]` and `[TestCase("DefBool")]`, etc., then define it as `DefType(string defType)` and string-interpolate the argument into the input code. Then you can have one test with all the def types.

Reply:
Your tests are all testing the functions. You should be testing the variables -- for example this:

```DefInt i
Dim i

Public Sub Foo()
  Debug.Print TypeName(i) 'Returns an integer
End Sub
```

Ditto for usages as a parameters. The inspection technically doesn't care for those stuff but you need to verify that it won't yield a false positive in cases where there is no `DefType` but implicit `Dim x` without the `As` clause.